Code/Framework/AzFramework/AzFramework/Archive/NestedArchive.cpp

```diff
@@ -89,11 +89,52 @@ namespace AZ::IO
         return m_pCache->RemoveDir(fullPath);
     }
 
+    //////////////////////////////////////////////////////////////////////////
     int NestedArchive::RemoveAll()
     {
         return m_pCache->RemoveAll();
     }
 
+    //////////////////////////////////////////////////////////////////////////
+    // Helper for 'ListAllFiles' to recursively traverse the FileEntryTree and gather all the files
+    void EnumerateFilesRecursive(AZ::IO::Path currentPath, ZipDir::FileEntryTree* currentTree, AZStd::vector<AZStd::string>& fileList)
+    {
+        // Drill down directories first...
+        for (auto dirIter = currentTree->GetDirBegin(); dirIter != currentTree->GetDirEnd(); ++dirIter)
+        {
+            if (ZipDir::FileEntryTree* subTree = currentTree->GetDirEntry(dirIter);
+                subTree != nullptr)
+            {
+                EnumerateFilesRecursive(currentPath / currentTree->GetDirName(dirIter), subTree, fileList);
+            }
+        }
+
+        // Then enumerate the files in current directory...
+        for (auto fileIter = currentTree->GetFileBegin(); fileIter != currentTree->GetFileEnd(); ++fileIter)
+        {
+            AZ::IO::Path filePath = currentPath / currentTree->GetFileName(fileIter);
+            fileList.emplace_back(filePath.c_str());
+        }
+    }
+
+    //////////////////////////////////////////////////////////////////////////
+    // lists all files in the archive
+    int NestedArchive::ListAllFiles(AZStd::vector<AZStd::string>& fileEntries)
+    {
+        AZStd::vector<AZStd::string> filesInArchive;
+
+        ZipDir::FileEntryTree* tree = m_pCache->GetRoot();
+        if (!tree)
+        {
+            return ZipDir::ZD_ERROR_UNEXPECTED;
+        }
+
+        EnumerateFilesRecursive({ "", AZ::IO::PosixPathSeparator }, tree, filesInArchive);
+
+        AZStd::swap(fileEntries, filesInArchive);
+        return ZipDir::ZD_ERROR_SUCCESS;
+    }
```

Comment:
The `EnumerateFilesRecursive` function call can just pass in fileEntries variable directly.
There is no need for a temporary vector that gets swappwed

```suggestion
    int NestedArchive::ListAllFiles(AZStd::vector<AZ::IO::Path>& fileEntries)
    {
        ZipDir::FileEntryTree* tree = m_pCache->GetRoot();
        if (!tree)
        {
            return ZipDir::ZD_ERROR_UNEXPECTED;
        }

        EnumerateFilesRecursive({ "", AZ::IO::PosixPathSeparator }, tree, filesInArchive);

        return ZipDir::ZD_ERROR_SUCCESS;
    }
```

Reply:
I just did that in order to "clear" out the argument of any data it may have had when called.